dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -250,13 +256,34 @@ private String readFileFromRepo(String fileName, String reference, GHRepository
                 start.add(folders.get(i));
                 String partialPath = Joiner.on("/").join(start);
                 try {
-                    Pair<GHContent, String> innerContent = getContentAndMetadataForFileName(partialPath, reference, repo);
-                    if (innerContent != null && innerContent.getLeft().getType().equals("symlink")) {
-                        // restart the loop to look for symbolic links pointed to by symbolic links
-                        List<String> newfolders = Lists.newArrayList(innerContent.getRight().split("/"));
-                        List<String> sublist = folders.subList(i + 1, folders.size());
-                        newfolders.addAll(sublist);
-                        folders = newfolders;
+                    Pair<GHContent, String> innerContent = getContentAndMetadataForFileName(partialPath, reference, repo, submoduleRedirected);
+                    if (innerContent != null) {
+                        if (innerContent.getLeft().getType().equals("symlink")) {
+                            // restart the loop to look for symbolic links pointed to by symbolic links
+                            List<String> newfolders = Lists.newArrayList(innerContent.getRight().split("/"));
+                            List<String> sublist = folders.subList(i + 1, folders.size());
+                            newfolders.addAll(sublist);
+                            folders = newfolders;
+                        } else if (innerContent.getLeft().getType().equals("submodule")) {
+                            String otherRepo = innerContent.getLeft().getGitUrl();
+                            if (otherRepo == null) {
+                                // likely means this submodule is not on GitHub, rest API reports it as null
+                                LOG.warn("Could not process %s at %s, is likely a submodule that is not on GitHub".formatted(originalFileName, originalReference));
+                                return null;
+                            }
+                            URL otherRepoURL = new URL(otherRepo);
+                            // reassign repo and reference
+                            final String[] split = otherRepoURL.getPath().split("/");
+                            final int indexPastReposPrefix = 2;
+                            String newRepositoryId = split[indexPastReposPrefix] + "/" + split[indexPastReposPrefix + 1];
```

Comment:
Are we sure that `split` will always have an element with index `2`?

Reply:
Yeah, the `otherRepoURL.getPath()` looks like `/repos/dockstore-testing/wdl-common/git/trees/165cc28fa8acc0f7d2a05e2b66082b5c95fe5043` 
2 only gets you to `dockstore-testing`